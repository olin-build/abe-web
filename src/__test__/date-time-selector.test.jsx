import React from 'react';
import moment from 'moment';
import renderer from 'react-test-renderer';
import DateTimeSelector from '../components/date-time-selector';

describe('DateTimeSelector', () => {
  test('datetime', () => {
    const jsx = (
      <DateTimeSelector datetime={moment('2018-05-14T12:00:00Z')} onChange={() => undefined} />
    );
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('date', () => {
    const jsx = (
      <DateTimeSelector
        datetime={moment('2018-05-14T12:00:00Z')}
        show="date"
        onChange={() => undefined}
      />
    );
    const component = renderer.create(jsx);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
