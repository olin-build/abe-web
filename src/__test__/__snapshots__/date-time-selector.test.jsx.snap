// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DateTimeSelector date 1`] = `
<div
  className="date-time-container"
>
  <button
    className="button"
    onClick={[Function]}
  >
    May 14, 2018
  </button>
  <div
    className="picker-popup-container"
  />
</div>
`;

exports[`DateTimeSelector datetime 1`] = `
<div
  className="date-time-container"
>
  <button
    className="button"
    onClick={[Function]}
  >
    May 14, 2018 8:00 AM
  </button>
  <div
    className="picker-popup-container"
  />
</div>
`;
